corehq/apps/app_manager/suite_xml/post_process/menu.py

```diff
@@ -1,28 +1,38 @@
-"""
-GridMenuHelper
------------------
-
-This is pretty simple: it just adds a root menu if one isn't already present.
-"""
 from corehq.apps.app_manager import id_strings
 from corehq.apps.app_manager.suite_xml.contributors import PostProcessor
+from corehq.apps.app_manager.suite_xml.sections.entries import EntriesHelper
 from corehq.apps.app_manager.suite_xml.xml_models import Menu, Text
 from corehq.util.timer import time_method
 
 
+def _get_root_menu(menus):
+    for menu in menus:
+        if menu.id == id_strings.ROOT:
+            return menu
+
+
 class GridMenuHelper(PostProcessor):
+    """This is pretty simple: it just adds a root menu if one isn't already present."""
 
     @time_method()
     def update_suite(self):
-        """
-        Ensure that there exists at least one menu where id="root".
-        """
-        if not self._contains_root_menu(self.suite.menus):
+        """Ensure that there exists at least one menu where id="root"."""
+        if not _get_root_menu(self.suite.menus):
             self.suite.menus.append(Menu(id=id_strings.ROOT, text=Text(), style="grid"))
 
-    @staticmethod
-    def _contains_root_menu(menus):
-        for menu in menus:
-            if menu.id == id_strings.ROOT:
-                return True
-        return False
+
+class RootMenuAssertionsHelper(PostProcessor):
+    """Set app-level custom assertions on root menu, creating one if necessary"""
```

Comment:
I believe I could've also made this always add a new root menu with just the assertions, as commcare combines multiple menus with the same ID.  Not sure it matters much one way or the other.

Reply:
Confirming you _could_ have done that, but that's not to say you _should_ have, I don't have strong feelings.